clients/src/main/java/org/apache/kafka/common/errors/PendingTransactionException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.errors;
+
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.Set;
+
+/**
+ * Exception thrown when there is pending transactional offset commit.
+ */
+public class PendingTransactionException extends RetriableException {
+
+    private static final long serialVersionUID = 1L;
+
+    public PendingTransactionException(String message) {
+        super(message);
+    }
+
+    public PendingTransactionException(Set<TopicPartition> pendingTopicPartitions) {
```

Comment:
I'd suggest not exposing any non-standard exception constructors. It is difficult to change them to add additional detail.